### PR TITLE
[L] remove user model

### PR DESCRIPTION
**Pull request by juanignaciosl**
~~PoC for removing old `User` model.~~

Small refactor with extractions from `User` model.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@gfiorav , @javitonino could you both take a look at this, please? Not because of being a risky PR (it is not) but for a couple of details. I've extracted a module and then included it back with `extend`, which is something that we don't usually do but enables small steps.

File: app/models/carto/user.rb [hunk 1]
```diff
@@ -512,4 +515,22 @@ def new_visualizations_version
   def engine_enabled?
     has_organization? ? organization.engine_enabled : engine_enabled
   end
+
+  def can_change_email?
+    (!google_sign_in || last_password_change_date.present?) && !Carto::Ldap::Manager.new.configuration_present?
+  end
+
+  def can_change_password?
+    !Carto::Ldap::Manager.new.configuration_present?
```
Comment (javitonino) [inline app/models/carto/user.rb]: As a note, `configuration_present?` should probably be a class method. Not related to this PR though.

File: lib/carto/user_authenticator.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module Carto
```
Comment (javitonino) [inline lib/carto/user_authenticator.rb]: I think you also intended to reuse this from `Carto::UserService`, but I cannot see it anywhere in the PR.

File: spec/factories/users.rb [hunk 3]
```diff
@@ -78,6 +77,10 @@
       CartoDB::UserModule::DBService.any_instance.stubs(:enable_remote_db_user).returns(true)
     end
 
+    after(:build) do |carto_user|
+      carto_user.crypted_password = ::User.password_digest(carto_user.password, carto_user.salt)
```
Comment (javitonino) [inline spec/factories/users.rb]: Nice!

File: app/models/carto/user.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/models/carto/user.rb]: I prefer to avoid changing anything outside `User` in this PR.

File: lib/carto/user_authenticator.rb (hunk 2 above)
Comment (juanignaciosl) [inline lib/carto/user_authenticator.rb]: I gave this a second thought and finally opted with removing usages of User. Stay tuned for 🍏  :-)

**Comment by juanignaciosl** [comment]
Next CR, please? :-)

File: spec/services/carto/user_authenticator_spec.rb [hunk 4]
```diff
@@ -2,6 +2,8 @@
 require 'carto/user_authenticator'
 
 describe Carto::UserAuthenticator do
+  include Carto::UserAuthenticator
```
Comment (javitonino) [inline spec/services/carto/user_authenticator_spec.rb]: 😮

**Pull request merged**